spring/boot-autoconfigure/src/main/java/com/linecorp/armeria/spring/ArmeriaBeanPostProcessor.java

```diff
@@ -0,0 +1,214 @@
+ * by default, {@link LocalArmeriaPort} and {@link LocalArmeriaPorts} annotations.
+ *
+ * <p>This class was created by referring to classes implementing
+ * Spring's {@link InstantiationAwareBeanPostProcessor}.
+ */
+public final class ArmeriaBeanPostProcessor implements InstantiationAwareBeanPostProcessor, Ordered {
+
+    private final Map<String, InjectionMetadata> injectionMetadataCache = new ConcurrentHashMap<>(256);
+
+    private final BeanFactory beanFactory;
+
+    /**
+     * Creates a new instance.
+     */
+    public ArmeriaBeanPostProcessor(BeanFactory beanFactory) {
+        this.beanFactory = requireNonNull(beanFactory, "beanFactory");
+    }
+
+    @Override
+    public int getOrder() {
+        return Ordered.LOWEST_PRECEDENCE - 2;
+    }
+
+    /**
+     * Post-processes the given property values before the factory apply them to the given bean.
+     */
+    @Override
+    public PropertyValues postProcessProperties(PropertyValues pvs, Object bean, String beanName) {
+        final InjectionMetadata metadata = findLocalArmeriaPortMetadata(beanName, bean.getClass(), pvs);
+        try {
+            metadata.inject(bean, beanName, pvs);
+        } catch (Throwable ex) {
+            throw new BeanCreationException(beanName, "Injection of resource dependencies failed", ex);
+        }
+        return pvs;
+    }
+
+    private InjectionMetadata findLocalArmeriaPortMetadata(
+            String beanName, Class<?> clazz, @Nullable PropertyValues pvs) {
+        final String cacheKey = StringUtils.hasLength(beanName) ? beanName : clazz.getName();
+        InjectionMetadata metadata = injectionMetadataCache.get(cacheKey);
+        if (InjectionMetadata.needsRefresh(metadata, clazz)) {
+            synchronized (injectionMetadataCache) {
+                metadata = injectionMetadataCache.get(cacheKey);
+                if (InjectionMetadata.needsRefresh(metadata, clazz)) {
+                    if (metadata != null) {
+                        metadata.clear(pvs);
+                    }
+                    metadata = buildLocalArmeriaPortMetadata(clazz);
+                    injectionMetadataCache.put(cacheKey, metadata);
+                }
+            }
+        }
+        return metadata;
+    }
+
+    private InjectionMetadata buildLocalArmeriaPortMetadata(Class<?> clazz) {
+        final List<InjectedElement> elements = new ArrayList<>();
+        Class<?> targetClass = clazz;
+
+        do {
+            final List<InjectionMetadata.InjectedElement> currElements = new ArrayList<>();
+
+            ReflectionUtils.doWithLocalFields(targetClass, field -> {
+                if (field.isAnnotationPresent(LocalArmeriaPort.class)) {
+                    if (Modifier.isStatic(field.getModifiers())) {
+                        throw new IllegalStateException(
+                                "LocalArmeriaPort annotations are not supported on static fields");
```

Comment:
This is a field so I think we need to use a singular sentence.
`"LocalArmeriaPort annotation is not supported on the static field: " + field.getName()`

Reply:
Fixed 568291b